pytorch_lightning/callbacks/gpu_stats_monitor.py

```diff
@@ -154,23 +157,27 @@ def _get_gpu_stats(self, gpu_stat_keys):
             check=True
         )
 
-        def _to_float(x):
+        def _to_float(x: str) -> float:
             try:
                 return float(x)
             except ValueError:
                 return 0.
 
         stats = result.stdout.strip().split(os.linesep)
-        stats = [list(map(_to_float, x.split(', '))) for x in stats]
+        stats = [[_to_float(x) for x in s.split(', ')] for s in stats]
+        return stats
 
+    @staticmethod
+    def _parse_gpu_stats(gpu_ids: str, stats: List[List[float]], keys: List[Tuple[str, str]]) -> Dict[str, float]:
+        """Parse the gpu stats into a loggable dict"""
         logs = {}
-        for i, gpu_id in enumerate(self._gpu_ids.split(',')):
-            gpu_stat_keys = [f'gpu_id: {gpu_id}/{x} ({unit})' for x, unit in gpu_stat_keys]
-            logs.update(dict(zip(gpu_stat_keys, stats[i])))
-
+        for i, gpu_id in enumerate(gpu_ids.split(',')):
+            for j, (x, unit) in enumerate(keys):
```

Comment:
I remember changing it to zip and updated a few more things to make it log faster because it was comparably slow(not too much) with inner for loop. Can you compare it again please?

Reply:
Seems to be the opposite:

```python
N = 10
M = 5

stats = [list(range(M)) for _ in range(N)]
keys = [(i, i) for i in range(M)]

def no_zip():
    logs = {}
    for i, gpu_id in enumerate(range(N)):
        for j, (x, unit) in enumerate(keys):
            logs[f'gpu_id: {gpu_id}/{x} ({unit})'] = stats[i][j]
    return logs

def with_zip():
    logs = {}
    for i, gpu_id in enumerate(range(N)):
        k = [f'gpu_id: {gpu_id}/{x} ({unit})' for x, unit in keys]
        logs.update(dict(zip(k, stats[i])))
    return logs

if __name__ == '__main__':
    import timeit
    assert no_zip() == with_zip()
    print("With zip:", timeit.timeit("with_zip()", setup="from __main__ import with_zip"))
    print("No zip:", timeit.timeit("no_zip()", setup="from __main__ import no_zip"))
```

```
With zip: 23.294362050997734
No zip: 18.493936623999616
```

Reply:
Ok, maybe something else was slowing down that before.

Reply:
I am not familiar with this particular problem you are discussion here, but just remember that if you benchmark something by measuring the execution time, make sure to run it for several trials and average the results. It's probably fine though.

Reply:
`timeit.timeit` does it automatically for a million executions

Reply:
😅 nice!